tron/core/job.py

```diff
@@ -277,12 +275,23 @@ def __init__(self, job):
         self.shutdown_requested = False
         self.watch(job)
 
-    def restore_state(self, job_state_data):
+    def restore_state(self, job_state_data, config_action_runner):
         """Restore the job state and schedule any JobRuns."""
-        self.job.restore_state(job_state_data)
+        job_runs = self.job.get_job_runs_from_state(job_state_data)
+        for run in job_runs:
+            self.job.watch(run)
+        self.job.runs.runs.extend(job_runs)
```

Comment:
i know we're single-threaded right now, but is this thread safe? i imagine we will eventually parallelize some parts of tron and this line kinda scares me :)

Reply:
probably not. but this isn't new code - just shuffled around.